geth/node/utils.go

```diff
@@ -10,16 +10,16 @@ import (
 // HaltOnPanic recovers from panic, logs issue, sends upward notification, and exits
 func HaltOnPanic() {
 	if r := recover(); r != nil {
-		err := fmt.Errorf("%v: %v", ErrNodeRunFailure, r)
+		strErr := fmt.Sprintf("%v: %v", ErrNodeRunFailure, r)
 
 		// send signal up to native app
 		signal.Send(signal.Envelope{
 			Type: signal.EventNodeCrashed,
 			Event: signal.NodeCrashEvent{
-				Error: err.Error(),
+				Error: strErr,
 			},
 		})
 
-		common.Fatalf(err) // os.exit(1) is called internally
+		common.Fatalf(ErrNodeRunFailure, r) // os.exit(1) is called internally
```

Comment:
I know it was there. But I really not sure if we should use `common.Fatalf` here. @adambabik @divan what's your opinion?

Reply:
@mandrigin i am changing this, it will be reported as a regular error to a caller of status backend

Reply:
So, @dshulyak if you are about to change this there are no problems to rebase to presented changes since the code doesn't really change the interaction interface. AFAIU, there's no particular blockers.

Reply:
it was an answer to a question about fatalf usage. i wasn't implying that these changes are giving me any additional problems